src/setup_payload/Base38Decode.h

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Functions to decode a Base38 String
+ */
+
+#pragma once
+
+#include "Base38.h"
+
+#include <string>
+#include <vector>
+
+namespace chip {
+
+// returns CHIP_NO_ERROR on successful decode
```

Comment:
delete this comment: it is obvious that returning CHIP_ERROR means 'return CHIP_NO_ERROR on success and an error code on failure)